projection/core/src/main/scala/com/lightbend/lagom/internal/projection/WorkerHolderActor.scala

```diff
@@ -0,0 +1,153 @@
+
+  def props(
+      projectionName: String,
+      workerProps: WorkerCoordinates => Props,
+      projectionRegistryActorRef: ActorRef
+  ): Props = {
+
+    val backOffChildProps: WorkerCoordinates => Props = coordinates =>
+      BackoffSupervisor.props(
+        BackoffOpts.onFailure(
+          workerProps(coordinates),
+          childName = coordinates.workerActorName,
+          minBackoff = 3.seconds,
+          maxBackoff = 30.seconds,
+          randomFactor = 0.2
+        )
+      )
+
+    /*
+     * <pre>
+     * WorkerHolder uses names provided by Akka Cluster sharding (we don't control that name)
+     *   +-- BackOffActor (named `coordinates.backoffActorName` as set on `doStart` below)
+     *      +-- actual worker Actor (named `coordinates.workerActorName` as set on in lines above)
+     * </pre>
+     */
+
+    Props(
+      new WorkerHolderActor(
+        projectionName,
+        backOffChildProps,
+        projectionRegistryActorRef
+      )
+    )
+  }
+
+}
+
+class WorkerHolderActor(
+    projectionName: String,
+    workerProps: WorkerCoordinates => Props,
+    projectionRegistryActorRef: ActorRef
+) extends Actor
+    with ActorLogging {
+
+  override def receive: Receive = unidentified
+
+  var lastStashed: Option[Status] = None
+
+  private def unidentified: Receive = {
+    case EnsureActive(tagName) =>
+      val coordinates = WorkerCoordinates(projectionName, tagName)
+      log.debug(s"Requesting registry of $coordinates [${self.path.toString}].")
+      projectionRegistryActorRef ! ProjectionRegistryActor.RegisterProjectionWorker(coordinates)
+      // become stopped and await for instructions from the Registry
+      becomeStopped(coordinates)
+  }
+
+  // the response to a RegisterProjection in `projectionRegistryActorRef` should be the
+  // requested Status which we will handle here.
+  // a) if the requested status has been set and propagated to our node then we'll get the value
+  // b) if the requested status is still not set in this node for our workerName then we won't
+  //    get anything as a response. But! Anytime a change on the requested status is detected, that
+  //    is propagated to the appropriate worker (aka, this) so the requested status will eventually
+  //    arrive here.
+  // c) TODO:: (potential improvement in case of edge cases not covered in (a) and (b)) we can
+  //    even send a new `RegisterProjection` from the `identified` behaviour anytime an
+  //    `EnsureActive` arrives. This will force a redelivery of the requested status and will
+  //    decouple the implementation of ProjectionRegistryActor from this implementation.
+
+  private def started(coordinates: WorkerCoordinates): Receive = {
+    case EnsureActive(_) => // yes, we're active
+    case Started         => // yes, we're started
+    case Stopped         => doStop(coordinates)
+  }
+
+  private def stopped(coordinates: WorkerCoordinates): Receive = {
+    case EnsureActive(_) => // yes, we're active
```

Comment:
For the future: I've found the mix of _command_ (ensure active) and _status_ (stopped) a little confusing here.

Reply:
Aaah, I like this change! Let's do on a separate PR, though. Something like:

```scala
    case EnsureActive(_)        => 
    case StateRequest(Stopped)  =>
    case StateRequest(Started)  =>
```
Instead of the actual

```scala
    case EnsureActive(_)  => 
    case Stopped          =>
    case Started          =>
```
makes it a lot clearer that we're receiving a demand/command and not reacting to a change.